### PR TITLE
auto-update medicare HICN to MBI

### PR DESCRIPTION
**Pull request by stephenwaite**
get mbi from era, update policy number in insurance data if globals m…bi update is set

this will be handy for billers so the medicare policy # will auto update when they post an ERA, will start debug/test tomorrow

### CODE REVIEW

File: library/billing/src/ParseERA.php [hunk 1]
```diff
@@ -310,10 +313,14 @@ public static function parse_era($filename, $cb)
                 $out['provider_fname'] = trim($seg[4]);
                 $out['provider_mname'] = trim($seg[5]);
                 $out['provider_member_id'] = trim($seg[9]);
-            } else if ($segid == 'NM1' && $seg[1] == 'TT' && $out['loopid'] == '2100') {
+            } // TT = Crossover Carrier (Transfer To another payer)
+            else if ($segid == 'NM1' && $seg[1] == 'TT' && $out['loopid'] == '2100') {
                 $out['crossover'] = 1;//Claim automatic forward case.
             } // 74 = Corrected Insured
-            // TT = Crossover Carrier (Transfer To another payer)
+            else if ($segid == 'NM1' && $seg[1] == '74' && $out['loopid'] == '2100') {
```
Comment (bradymiller) [inline library/billing/src/ParseERA.php]: PSR2 will require `} else if {` so gonna need to reposition the comments

File: library/billing/src/ParseERA.php [hunk 2]
```diff
@@ -310,10 +313,14 @@ public static function parse_era($filename, $cb)
                 $out['provider_fname'] = trim($seg[4]);
                 $out['provider_mname'] = trim($seg[5]);
                 $out['provider_member_id'] = trim($seg[9]);
-            } else if ($segid == 'NM1' && $seg[1] == 'TT' && $out['loopid'] == '2100') {
+            } // TT = Crossover Carrier (Transfer To another payer)
+            else if ($segid == 'NM1' && $seg[1] == 'TT' && $out['loopid'] == '2100') {
```
Comment (bradymiller) [inline library/billing/src/ParseERA.php]: PSR2 will require `} else if {` so gonna need to reposition the comments

**Comment by bradymiller** [comment]
Code looks good to me.

File: interface/billing/sl_eob_process.php [hunk 3]
```diff
@@ -621,6 +622,28 @@ function era_callback(&$out)
...
             }
+
+            if ($out['corrected'] == '1') {
+                if ($GLOBALS['update_mbi']) {
+                    if ($primary) {
+                        InsuranceService::update($pid, "primary", array($insurance_id, $out['corrected_mbi']));
+                    } else { // secondary medicare
+                        InsuranceService::update($pid, "secondary", array($insurance_id, $out['corrected_mbi']));
+                    }
+
```
Comment (stephenwaite) [inline interface/billing/sl_eob_process.php]: looks like this function needs every data element provided, would it be ok to add a function to the class that sets only one of the fields?

Comment (stephenwaite) [reply interface/billing/sl_eob_process.php]: going to keep debugging to amke sure $insurance_id is set

**Comment by stephenwaite** [comment]
testing well :)

**Comment by bradymiller** [comment]
hi @stephenwaite , Just let me know when this is ready to come into codebase. -brady

**Comment by stephenwaite** [comment]
it's ready, thank you @bradymiller

**Comment by bradymiller** [comment]
hi @stephenwaite , Brought this in. Thanks for the code improvements! -brady

**Pull request merged**